### PR TITLE
Submitted the solutions of week02's problems.

### PR DESCRIPTION
**Pull request by saifuzzaman-saif**
There was a mistake in the week01's PR. Now I've fixed that. I want that PR to be reviewed for letting me know if there are scopes to improve in the coding style or anything else. The previous week's PR link : https://github.com/saifuzzaman-saif/tjip-leetcode-repo/pull/2

Solved the following problems in this PR :

Longest Substring Without Repeating Characters
String to Integer (atoi)
Longest Common Prefix
Valid Parentheses
Implement strStr()
Group Anagrams
Minimum Window Substring
Largest Number
Valid Anagram
Longest Substring with At Least K Repeating Characters
Text Justification
Decode String
Number of Matching Subsequences
Find and Replace In String

### CODE REVIEW

File: week02/GroupAnagrams.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            List<String>list = anagrams.getOrDefault(hashVal, new ArrayList<String>());
+            list.add(str);
+            anagrams.put(hashVal, list);
+        }
+
+        return new ArrayList<>(anagrams.values());
+    }
+
+    private int getHashVal(String s) {
+        long hash = 1, offset = 997, mod = 1_000_000_007;
```
Comment (Sadman007) [inline week02/GroupAnagrams.java:21]: Using `int` as the hash data type is extremely dangerous with such a low mod value. The return type of `getHashVal` should be `long`.

File: week02/ValidAnagram.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    private int getHash(String s) {
+        long hash = 1, offset = 997, mod = 1_000_000_007;
+
+        for (int i = 0; i < s.length(); i++) {
+            hash = (hash * (offset + s.charAt(i))) % mod;
+        }
+
+        return (int) hash;
+    }
```
Comment (Sadman007) [inline week02/ValidAnagram.java]: Same as above comment.

File: week02/ValidParentheses.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                return false; 
+            } else if (ch == ')' && stk.peek() != '(') {
+                return false;
+            } else if (ch == '}' && stk.peek() != '{') {
+                return false;
+            } else if (ch == ']' && stk.peek() != '[') {
+                return false;
+            } else {
+                stk.pop();
+            }
```
Comment (Sadman007) [inline week02/ValidParentheses.java]: Not a good way to solve this problem. What if I introduce a new kind of parentheses? You need to rewrite a lot of code again. Watch out the official solution as soon as I post them.

**Review by Sadman007 — APPROVED** [review]
Hi @saifuzzaman-saif 

Feel free to merge this PR. Thanks. Nice work.

File: week02/GroupAnagrams.java (hunk 1 above)
Comment (saifuzzaman-saif) [reply week02/GroupAnagrams.java:21]: Fixed.

File: week02/ValidAnagram.java (hunk 2 above)
Comment (saifuzzaman-saif) [reply week02/ValidAnagram.java]: Fixed

File: week02/ValidParentheses.java (hunk 3 above)
Comment (saifuzzaman-saif) [reply week02/ValidParentheses.java]: Refactored as needed.

**Pull request merged**